addons/account/models/account_journal_dashboard.py

```diff
@@ -682,15 +682,14 @@ def open_action(self):
 
         # Override the domain only if the action was not explicitly specified in order to keep the
         # original action domain.
+        action['domain'] = ast.literal_eval(action['domain'] or '[]')
```

Comment:
You need to check that `action['domain']` is a string (and not a list) before calling `literal_eval()`, otherwise it will raise a `ValueError` :wink:

Reply:
thanks for the feedback @epictete ,
but in the action['domain'] we will have only a valid domain as List or False right, just trying to understand when we will have a case of a string

also i believe when it is false it is handled by the empty list passed in the updated code

Reply:
The error must be that when reading the action through the xmlid, the function "_get_action_dict", does not transform the string saved in the database to a list, it is true that you make the check when creating but when reading I do not see anything that transforms the string into a list.

Reply:
@AntoniRomera do you have suggestion in the code level that i can make

Reply:
In the model 'ir.actions.act_window' extending from "ir.actions.actions", call the parent function and if the domain is a string evaluate it. An example of the following:
```
def _get_action_dict(self):
    """ Returns the action content for the provided action record.
    """
    self.ensure_one()
    action = super()._get_action_dict()
    if action.get("domain") and isinstance(action.get("domain"), str):
        action["domain"] = safe_eval(action.get("domain"))
    return action
```
Another example would be to add the following lines to the "read" function within the "r.actions.act_window" model.
```
....
for values in result:
    if values.get("domain") and isinstance(values.get("domain"), str):
        values["domain"] = safe_eval(values.get("domain"))
return result
```

If we wanted to go deeper we would have to look at the original "read" function and somehow mark all the fields that are domains and make it return a list.

Reply:
@AntoniRomera thanks, but it seems the literal_eval is handling the string and converting it into list

Reply:
Of course, but if you want to make a more correct proposition and not only fix this function but all the possible bugs it may cause and save everyone from having to do a literal_eval every time, it would be better to fix at source. Because the error is given by calling this function "_for_xml_id" I think it is a very common function.

Reply:
thanks @AntoniRomera  updated the PR as you have suggested

@epictete if you have time, can you review